_docs/step-development-guideline.md

```diff
@@ -1,5 +1,45 @@
 # Step Development Guideline
 
+## Step dependencies
+
+In the context of Steps, dependency means an executable available in the PATH environment variable, which is executed by the step.
+
+`deps` should not include an executable needed to run the step.
+
+_For example `go` should not be listed as a step dependency in case of a step written in go, but needs to be listed if the step calls the `go` binary as the [Go test step](https://github.com/bitrise-steplib/steps-go-test/blob/master/main.go#L86) does._
+
+A step dependency is installed by the `Bitrise CLI` if it is not available in the PATH environment variable. Unused dependency increases the build time unnecessarily, avoid listing a dependency which is not executed by the step.
+
+Since steps can be performed in any environment where Bitrise CLI can run, list every used dependency, even if you know that they are pre-installed on the Bitrise build virtual machines.
+
+Bitrise CLI can install step dependencies available in Homebrew package manager:
+
+```
+deps:
+  brew:
+  - name: cmake
+```
+
+apt-get dependencies available in sources listed in `sources.list` file on the host machine:
+
+```
+deps:
+  apt_get:
+  - name: cmake
+```
+
+Steps can define dependencies which needs to be available on the host machine but can not be installed on easy way (like using a single brew or apt-get command):
+
+```
+deps:
+  check_only:
+  - name: xcode
+```
+
+Currently the only one `check_only` dependency is `xcode`.
+
+Other dependencies needs to be installed and check during the step run time or using other steps.
+
```

Comment:
I have a missing info about if is the CLI still installs the tool if it is available under the PATH?

Reply:
What do you mean by this? Is not it the answer for this: https://github.com/bitrise-io/bitrise/pull/649/files#diff-acb597ec81b4cb89862c1706d0f368eaR11

Reply:
Oh yeah. Sure. Just wanted to make sure that my locally installed one will be untouched in case it is listed as a dep.